pychromecast/__init__.py

```diff
@@ -171,8 +179,23 @@ def __init__(self, host, port=None, device=None, **kwargs):
         self.status = None
         self.status_event = threading.Event()
 
+        self.browser = None
+        self.tracker = None
+
+        self.setup(host, port, self.device, self.kwargs)
+
+    def setup(self, host, port, device, kwargs):
+        """
+        Method for setting up, and resetting the chromecast object
+        """
+        tries = 2 if self.cast_type == CAST_TYPE_GROUP \
```

Comment:
Tries is `None` by default for all other `CAST_TYPES` meaning that the socket client will try to reconnect forever. By defining a finite number of tries we make sure that the socket client will eventually fail (when ip is switched) and then sends a `DISCONNECTED` event which we listen for.